change/@ni-nimble-components-df093e6a-7cba-46ed-8d5d-07670af90719.json

```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
+  "comment": "Add locale support to date-text column",
```

Comment:
This PR is not only adding locale support to the date-text column, it's adding locale support to the nimble-theme-provider as a whole. I think that's important to capture in the change file.